packages/h5p-server/src/ContentManager.ts

```diff
@@ -90,14 +96,27 @@ export default class ContentManager {
     }
 
     /**
-     * Deletes a piece of content and all files dependent on it.
+     * Deletes a piece of content, the corresponding contentUserData and all files dependent on it.
      * @param contentId the piece of content to delete
      * @param user the user who wants to delete it
      */
     public async deleteContent(
         contentId: ContentId,
         user: IUser
     ): Promise<void> {
+        if (this.contentUserDataStorage) {
```

Comment:
It might be sensible to acquire a lock that is also acquired when the user posts the content state. Otherwise a race condition is possible in which the content state is deleted, created again and then the content is deleted. The effect would be that there are orphaned content states. Alternatively it you could delete the content state after deleting the content.

Reply:
Ok, I will delete it after the content is deleted.